### DISCUSSION TITLE
Add a graph-based music synthetizer (similar to visual shaders)

### DISCUSSION THREAD

**Discussion by strawstack** [godotengine/godot-proposals#10416] [Engine Core]
### Describe the project you are working on

I’m building a 2D game. I love the VisualShader tools in Godot. I want to make ambient back ground music for my game. I would like a node graph based interface that would let me compose synth lofi style music that would loop in the background of my game

### Describe the problem or limitation you are having in your project

I want music, and I want to create the music from scratch inside the Godot application. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I would be able to easily add music to my game. The music could be procedural, endlessly looping, copy-write free, and space efficient. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Imagine the VisualShader interface but the nodes generate and manipulate music. 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

If GDScript could generate tones, I could make custom VisualShader nodes that emit or manipulate sound, and maybe the sound would play if the material was in my scene 

### Is there a reason why this should be core and not an add-on in the asset library?

VisualShaders are in core and are amazing for graphical capabilities. This proposed feature could be called VisualSound, and it would would do the same for sound and VisualShaders do for music. Godot would become even more of a one stop shop for all game dev needs

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot-proposals/issues/3394

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

Note that https://github.com/godotengine/godot-proposals/issues/3394 is close to what you want, but the nodes it intends to provide out of the box are not designed to create music. The same system could theoretically be used to create music, but it would require a *lot* of high-level tools added on top, which may be considered out of scope for the engine core (and better suited as an add-on).

This is a similar issue to what happened with visual scripting, which ended up being removed in the end since there weren't enough high-level nodes provided to really make it usable to non-programmers.
